### ISSUE TITLE
PlainTable with prefix_extractor do not support manual compaction?

### ISSUE THREAD

**Issue by Tostoyevsky** [facebook/rocksdb#9174]
use plaintable format with prefix_extractor & hash_table_ratio.
write data to db.
And I called CompactRange，but got a failed status:Invalid argument: total_order_seek not implemented for PlainTable.

why CompactRange should call PlainTableIterator::Seek method?
if it`s needed ,then why plaintable format don`t support Seek?

Thanks!
### Expected behavior

use plaintable format with prefix_extractor & hash_table_ratio.
CompactRange succeed.

### Actual behavior

got a failed status:Invalid argument: total_order_seek not implemented for PlainTable.

### Steps to reproduce the behavior
use plaintable format with prefix_extractor & hash_table_ratio.
write data to db, create 2 or more sst files.
called CompactRange (x, x , null, null);

**Comment by ajkr** [comment]
Do you use subcompactions by any chance? I can repro using that option (but not without it yet). It makes sense in the context of subcompactions because a subcompaction boundary may be in the middle of an SST and we need to seek to it.

**Comment by Tostoyevsky** [comment]
Yes, i set `CompactRangeOptions::max_subcompactions` > 1 to speed up manual compaction.
But plaintable sst file is ordered no matter with prefix or not.  then why `PlainTableIterator::Seek()`  with prefix is not supported?
what's the difference?
THX!

**Comment by VladRodionov** [comment]
No, this exception has nothing to do with max_subcompaction > 1. In my case, this config is 1 and it fails with the same exception. ??? Is PlainTable format really supported in Java?

**Comment by ajkr** [comment]
I see. Are you also talking about an error returned by `CompactRange()`? Are you able to provide a stack trace where the `NotSupported` error is coming from?
